### DISCUSSION TITLE
Guidance: Configuring second PnPContext in Blazor .Net7

### DISCUSSION THREAD

**Discussion by brmykleb** [pnp/pnpcore#1255] [Q&A] [answered]
I have a working solution exposing a subset of a sharepoint list (document library) to authenticated users in a blazor wasm app. I have used the custom IAuthenticationProvider approach from Demo.Blazor and set this up in the client project of my solution.
Now I want to use PnPContext to write data from a form input by unauthenticated users to a Teams chat. I suppose the best approach for configuring authentication for this second PnPcontext is using the certificate based approach.
Do you have recommendations as to where I should configure the second context?
Can I configure it in Program.cs in my client project and use DI? Could this cause issues resolving which context to use?? 
Or should I rather do it in Program.cs in the server project and call it via webapi?
I have searched through the other discussions, but it is not entirely clear to me what is the best approach, nor how I should configure two contexts in the same Program.cs.
Any advice is appreciated.

**Comment by jansenbe** [comment, marked as answer]
@brmykleb : I don't think you should do certificate based auth from blazor wasm as you'll have to include your cert which folks can reverse engineer. Especially important since you're using application (app-only) and as such expose access to a complete tenant. The recommended way for that is to have a service endpoint that you secure via Azure AD and you call that from your app. The service endpoint can then safely use application permissions with a cert to make changes
